WordPress/WordPressTest/ViewRelated/Post/Views/PostCardStatusViewModelTests.swift

```diff
@@ -0,0 +1,51 @@
+
+@testable import WordPress
+
+import Nimble
+
+private typealias ButtonGroups = PostCardStatusViewModel.ButtonGroups
+
+class PostCardStatusViewModelTests: XCTestCase {
+    private var contextManager: TestContextManager!
+    private var context: NSManagedObjectContext!
+
+    override func setUp() {
+        super.setUp()
+        contextManager = TestContextManager()
+        context = contextManager.newDerivedContext()
+    }
+
+    override func tearDown() {
+        context = nil
+        contextManager = nil
+        super.tearDown()
+    }
+
+    func testExpectedButtonGroupsForVariousPostAttributeCombinations() {
+        // Arrange
+        let expectations: [Post: ButtonGroups] = [
+            // Draft with remote
+            PostBuilder(context).drafted().withRemote().build():
+                ButtonGroups(primary: [.edit, .view, .more], secondary: [.publish, .trash]),
+            // Draft that was not uploaded to the server
+            PostBuilder(context).drafted().with(remoteStatus: .failed).build():
+                ButtonGroups(primary: [.edit, .publish, .trash], secondary: []),
+            // Local published draft with confirmed auto-upload.
+            PostBuilder(context).published().with(remoteStatus: .failed).confirmedAutoUpload().build():
+                ButtonGroups(primary: [.edit, .cancelAutoUpload, .more], secondary: [.moveToDraft, .trash]),
+            // Local published draft with canceled auto-upload.
+            PostBuilder(context).published().with(remoteStatus: .failed).build():
+                ButtonGroups(primary: [.edit, .publish, .more], secondary: [.moveToDraft, .trash]),
+            // Published post
+            PostBuilder(context).published().withRemote().build():
+                ButtonGroups(primary: [.edit, .view, .more], secondary: [.stats, .moveToDraft, .trash]),
+        ]
```

Comment:
I believe we can replace those comments in favor of multiple tests methods, what do you think?

This will make explicit which case is failing and why.

Reply:
Ok! My only concern is with tests that could possibly involve more logic and make it harder to understand them. We don't want to write tests for our tests, right? 🤣 

Anyway, this case seems pretty simple.